### PR TITLE
doc: correct reference links in the snapshot,release,upgrade docs

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: docs/ceph-csi-upgrade.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 - [Ceph-csi Upgrade](#ceph-csi-upgrade)
   - [Pre-upgrade considerations](#pre-upgrade-considerations)
-    - [snapshot-controller and snapshot Beta CRD](#snapshot-controller-and-snapshot-beta-crd)
+    - [snapshot-controller and snapshot Beta CRD](#snapshot-controller)
```
Comment (pkalever) [inline docs/ceph-csi-upgrade.md]: Should this be:
`s/[snapshot-controller and snapshot Beta CRD]/[snapshot-controller and snapshot CRD]/` ?

Comment (humblec) [reply docs/ceph-csi-upgrade.md]: yeah.. correcting it.

File: docs/snap-clone.md [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
+  - [Create CephFS Snapshot](#create-snapshot)
+  - [Restore CephFS Snapshot to a new PVC](#restore-snapshot-to-a-new-pvc)
+  - [Clone CephFS PVC](#clone-pvc)
 - [Create RBD Snapshot and Clone Volume](#create-rbd-snapshot-and-clone-volume)
   - [Create RBD SnapshotClass](#create-rbd-snapshotclass)
   - [Create RBD Snapshot](#create-rbd-snapshot)
-  - [Restore RBD Snapshot to a new PVC](#restore-rbd-snapshot-to-a-new-pvc)
-  - [Clone RBD PVC](#clone-rbd-pvc)
+  - [Restore RBD Snapshot to a new PVC](#restore-the-snapshot)
+  - [Clone RBD PVC](#clone-pvc)
```
Comment (Madhu-1) [inline docs/snap-clone.md]: @humblec Looks like few links are not working.

**Review by Madhu-1 — DISMISSED** [review]

File: docs/snap-clone.md (hunk 2 above)
Comment (humblec) [reply docs/snap-clone.md]: @Madhu-1 can you please list out ?

Comment (Madhu-1) [reply docs/snap-clone.md]: * clone rbd PVC is link point to cephfs clone

Any specif reason to remove cephfs and names from the heading? an we keep both cephfs and rbd heading similar?

now this is how it looks
## cephfs
* create-cephfs-snapshot-and-clone-volume
* create-cephfs-snapshotclass
* create-snapshot
* restore-snapshot-to-a-new-pvc
* clone-pvc

## rbd
* create-rbd-snapshot-and-clone-volume
* create-rbd-snapshotclass
* create-rbd-snapshot
* restore-the-snapshot
* clone-pvc

Comment (humblec) [reply docs/snap-clone.md]: @Madhu-1 thanks... done.. ptal

File: docs/ceph-csi-upgrade.md (hunk 1 above)
Comment (humblec) [reply docs/ceph-csi-upgrade.md]: @pkalever addressed .. thanks. PTAL

Comment (Madhu-1) [reply docs/ceph-csi-upgrade.md]: Is this not fixed yet? I still see beta CRD

**Comment by humblec** [comment]
@pkalever  @Madhu-1 @nixpanic ptal

**Review by pkalever — APPROVED** [review]
LGTM. Thanks!

File: docs/snap-clone.md [hunk 3]
```diff
@@ -4,12 +4,12 @@
 - [Create CephFS Snapshot and Clone Volume](#create-cephfs-snapshot-and-clone-volume)
   - [Create CephFS SnapshotClass](#create-cephfs-snapshotclass)
   - [Create CephFS Snapshot](#create-cephfs-snapshot)
-  - [Restore CephFS Snapshot to a new PVC](#restore-cephfs-snapshot-to-a-new-pvc)
+  - [Restore CephFS Snapshot to a new PVC](#restore-cephfs-snapshot)
   - [Clone CephFS PVC](#clone-cephfs-pvc)
```
Comment (pkalever) [inline docs/snap-clone.md:8]: @humblec can you also fix this link `Clone CephFS PVC`, I just realized its broken ?

File: docs/ceph-csi-upgrade.md [hunk 4]
```diff
@@ -52,7 +52,7 @@ fuse client or rbd-nbd as of now.
 This guide will walk you through the steps to upgrade the software in a cluster
 from v3.0 to v3.1
 
-### Snapshot-controller
+### Snapshot-controller-and-snapshot-crd
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md]:
```suggestion
### Snapshot-controller and snapshot-crd
````

File: docs/snap-clone.md (hunk 3 above)
Comment (humblec) [reply docs/snap-clone.md:8]: @pkalever sharp.. addressed in the latest patchset

File: docs/ceph-csi-upgrade.md (hunk 4 above)
Comment (humblec) [reply docs/ceph-csi-upgrade.md]: Done.. ptal @Madhu-1

**Review by Madhu-1 — COMMENTED** [review]
@humblec can you also change the below headings

* Create RBD-SnapshotClass to Create RBD SnapshotClass
* Create rbd-Snapshot ti Create RBD Snapshot

**Comment by humblec** [comment]
@Madhu-1 done.. thanks.

File: docs/snap-clone.md (hunk 3 above)
Comment (pkalever) [reply docs/snap-clone.md:8]: Thanks @humblec

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**